Adding extra metadata to module/components

alterx opened the pull request:
Issue:

## What I did
Added the ability to provide custom metadata to the story. This works, for example, when you need to include a service, a custom pipe or directive.

## How to test
`yarn bootstrap`
`cd examples/angular-cli`
`yarn storybook`

alterx committed:
Adding extra metadata to module/components

alterx committed:
Removing unused import

Signed-off-by: Carlos Vega <clmvega@gmail.com>

igor-dv reviewed (commented):
1. Is there some documentation that needs to be edited about how to use new features?
2. As one that is not familiar with most of the angular things (now better, but still =) ), I think it's very important to add descriptive and intuitive names of just everything. I would like to create some followup PR with a bit refactoring/renaming/separating things, will it be OK?

ndelangen commented on examples/angular-cli/src/stories/index.ts:
```diff
@@ -182,3 +164,63 @@ storiesOf('Addon Knobs', module)
+      name: 'Static name'
+    },
+    moduleMetadata: {
+      imports: [],
```
this is kinda documentation, but maybe add it to a real documentation too?

alterx replied:
Yeah, they're just there to show you can actually add all of them

ndelangen committed:
Merge branch 'release/3.3' into angular-module-metadata

alterx commented on app/angular/src/client/preview/angular/components/app.component.ts:
```diff
@@ -13,7 +13,8 @@ import {
+import { Data } from "../types";
 
 @Component({
   selector: "my-app",
```
We want these to be the same since they all hook into the same tag in the template.
I'm going to go with `app-root`

alterx commented on app/angular/src/client/preview/angular/components/error.component.ts:
```diff
@@ -1,5 +1,6 @@
+import { Data } from "../types";
 
 @Component({
   selector: "my-app",
```
We want these to be the same since they all hook into the same tag in the template.
I'm going to go with `app-root`

alterx commented on app/angular/src/client/preview/angular/helpers.ts:
```diff
@@ -2,7 +2,8 @@ import {
   NgModule,
   Component,
-  CUSTOM_ELEMENTS_SCHEMA
+  NgModuleRef,
```
Yeah, you're right, I probably screwed the `rebase` :P

alterx commented on examples/angular-cli/src/stories/index.ts:
```diff
@@ -20,28 +20,11 @@ import { Welcome, Button } from '@storybook/angular/demo';
 import { SimpleKnobsComponent } from './knobs.component';
```
Agreed. Back when we started with these the react example was this way, we surely need to update these too

alterx commented on examples/angular-cli/src/stories/moduleMetadata/service.component.ts:
```diff
@@ -0,0 +1,27 @@
+import { DummyService } from './dummy.service';
+
+@Component({
+  selector: 'simple-knobs-component',
```
Yup

ndelangen committed:
Merge branch 'release/3.3' into angular-module-metadata

alterx committed:
Igor's feedback

- Changing main tag to a more meaningful name
- Removing redundant type Data
- Validating moduleMetadata param
- Removing unused references

Signed-off-by: Carlos Vega <clmvega@gmail.com>

alterx committed:
Merge branch 'angular-module-metadata' of github.com:storybooks/storybook into angular-module-metadata

alterx commented:
@igor-dv after this one gets merged feel free to submit a PR to rename some stuff, these names were chosen on the go and now that we have a more feature complete product I think there's a lot of room for improvement in naming :) 

@ndelangen @igor-dv how do we want to procede with documentation?

noratodaitan commented:
I will try to help you, above all in docs.
I will use my personal profile @norato

igor-dv commented:
There is an outdated demo gif
![image](https://user-images.githubusercontent.com/7867954/34264322-35ec7d10-e67b-11e7-81c2-2098218354b5.png)
And it's used in all the apps... 😫

norato commented:
@igor-dv I can improve this gif

igor-dv commented:
@norato it will be very helpful

alterx commented:
Perfect, guys. Is any of these things happening in this PR? 🤔  @norato @igor-dv

igor-dv commented:
Nope. Just merge 👍
